File: test-suite/golden-tests/optional-r/foo.R
isNull <- function(x) {
    is.null(x)
}

fromNull <- function(default_val, x) {
    if (is.null(x)) return(default_val)
    return(x)
}

toNull <- function(x) {
    return(x)
}

safeHead <- function(xs) {
    if (length(xs) == 0) return(NULL)
    return(xs[[1]])
}

optionalAdd <- function(x, y) {
    if (is.null(x) || is.null(y)) return(NULL)
    return(x + y)
}

optionalList <- function(xs) {
    lapply(xs, function(x) if (x < 0) NULL else x)
}

countNulls <- function(xs) {
    sum(sapply(xs, is.null))
}

File: test-suite/golden-tests/optional-py/foo.py
def isNull(x):
    return x is None

def fromNull(default_val, x):
    if x is None:
        return default_val
    return x

def toNull(x):
    return x

def safeHead(xs):
    if len(xs) == 0:
        return None
    return xs[0]

def optionalAdd(x, y):
    if x is None or y is None:
        return None
    return x + y

def optionalList(xs):
    return [None if x < 0 else x for x in xs]

def countNulls(xs):
    return sum(1 for x in xs if x is None)
